### PR TITLE
REV-167-offers-table purchase counter offer forms

### PR DESCRIPTION
**Pull request by IbrahimAljabrST**
### Jira URL

https://optimumpartners.atlassian.net/browse/REV-167

### Description

add purchase offer modal

Dependencies in Other PRs,Repo
https://github.com/abukhadraop-or/reverifi-api/pull/62

### CODE REVIEW

File: src/components/listing-page/SubmitOffer/submit-offer.style.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  display: flex;
+  width: 50%;
+  height: 100%;
+  align-items: flex-start;
+  flex-direction: column;
+  padding: 1rem;
+`;
+
+export const Title = styled.h2`
+  font-weight: 600;
```
Comment (mohdabed123) [inline src/components/listing-page/SubmitOffer/submit-offer.style.js:28]:
```suggestion
```

File: src/components/listing-page/SubmitOffer/submit-offer.style.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+export const PriceContainer = styled.div`
+  padding-left: 1rem;
+`;
+
+export const Price = styled.h3`
+  margin: 0;
+  color: ${colors.green};
+  font-size: 1.5rem;
+  font-weight: 600;
```
Comment (mohdabed123) [inline src/components/listing-page/SubmitOffer/submit-offer.style.js:39]:
```suggestion
```

File: src/components/shared/ListingImageInput/index.jsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+ * Renders a form listing images input selector with image list
+ *
+ * @param  {boolean}     images        A list of selected images to display.
+ * @param  {Function}    onAddImages   Add images handler.
+ * @param  {JSX.Element} onDeleteImage Delete image handler.
+ * @param  {string}      label         Text for the
+ *
+ * @return {JSX.Element}
+ */
+function ListingImageInput({ images, onAddImages, onDeleteImage, label }) {
```
Comment (mohdabed123) [inline src/components/shared/ListingImageInput/index.jsx:24]: make this as shared component and use it ,

**Review by mohdabed123 — CHANGES_REQUESTED** [review]

File: src/components/dashboard/OffersTable/index.jsx [hunk 4]
```diff
@@ -25,85 +18,62 @@ import {
...
+          listingOffer.length > 0 && (
+            <TableRow key={id}>
+              <TableCell>
+                <Accordion title={address}>
+                  {listingOffer?.map(
+                    ({ price: offerPrice, attachments, offeredUser }) => (
+                      <OfferList>
+                        <OfferListItem>
+                          <OfferListItemText>Listing Price</OfferListItemText>
+                          <p>$ {price.toLocaleString()}</p>
```
[the remark this answers is not in the document]
Comment (IbrahimAljabrST) [reply src/components/dashboard/OffersTable/index.jsx:45]: we need toLocaleString to format the price

File: src/components/listing-page/Offer/index.jsx [hunk 5]
```diff
@@ -26,20 +28,30 @@ import React from 'react';
...
           </Location>
         </div>
         <SubmitOffer>
-          <SubmitButton aria-label="Submit an Offer" type="button">
+          <SubmitButton
+            aria-label="Submit an Offer"
+            type="button"
+            onClick={() => {
+              if (!isLoggedIn) navigate('/sign-up');
+              setModalData({ address, isShowOffer: true, price });
```
[the remark this answers is not in the document]
Comment (IbrahimAljabrST) [reply src/components/listing-page/Offer/index.jsx:52]: we will not show the modal if you are not logged in, and will navigate to sign up page

**Review by mohdabed123 — APPROVED** [review]

**Pull request merged**